src/flamenco/log_collector/fd_log_collector.h

```diff
@@ -400,37 +400,54 @@ fd_log_collector_program_success( fd_exec_instr_ctx_t * ctx ) {
 }
 
 /* fd_log_collector_program_success logs:
-     "Program <ProgramIdBase58> failed: <err>" */
+     "Program <ProgramIdBase58> failed: <err>"
+
+   This function handles the logic to log the correct msg, based
+   on the type of error (InstructionError, SyscallError...).
+   https://github.com/anza-xyz/agave/blob/v2.0.6/program-runtime/src/invoke_context.rs#L535-L549
+
+   The error msg is obtained by external functions, e.g. fd_vm_syscall_strerror(),
+   and can be either a valid msg or an empty string.  Empty string represents
+   special handling of the error log, for example the syscall panic logs directly
+   the result, and therefore can be skipped at this stage. */
 static inline void
 fd_log_collector_program_failure( fd_exec_instr_ctx_t * ctx ) {
   if( FD_LIKELY( ctx->txn_ctx->log_collector.disabled ) ) {
     return;
   }
 
-  /* TODO: we need to distinguish at least between:
-     - InstructionError
-     - EbpfError::SyscallError
-     - ProgramError?
-     https://github.com/anza-xyz/agave/blob/v2.0.6/program-runtime/src/invoke_context.rs#L535-L549 */
-  extern char const * fd_vm_strerror( int err );
+  extern char const * fd_vm_ebpf_strerror( int err );
+  extern char const * fd_vm_syscall_strerror( int err );
   extern char const * fd_executor_instr_strerror( int err );
 
```

Comment:
OOC: why not declare these in `fd_vm_base.h` (or a separate header)?

Reply:
I don't know, will give it a try... 🤞

Reply:
there's a circular dep. log is used by vm/exec, but this function needs them.
the core of the issue is how we defined instr_ctx, I think it's easier to address that in a separate PR.

Reply:
sounds good